### DISCUSSION TITLE
Gateway Util

### DISCUSSION THREAD

**Discussion by WASasquatch** [comfyanonymous/ComfyUI#439] [Ideas]
Similar to reroute, it accepts, and returns input, however, it does so between two nodes, Gateway In, and Gateway Out (probably both created at the same time from "Gateway" node.).

This would allow having really tidy workflows without wire mess from wrapping inputs behind it's output and off below somewhere, which casues line wires to warp all over behind your network. 

Ideally this would be a litegraph thing probably, so you could plug the gateway into the out gateway, and the node connection between becomes invisible. 

![image](https://user-images.githubusercontent.com/1151589/230660004-3f971252-6541-4f56-9230-ece8c2b612e8.png)

**Comment by omar92** [comment]
maybe they need tags

**Comment by omar92** [comment]
or better if it have link that can be visible or invisible

**Reply by WASasquatch** [reply]
What do you mean? The example image is just the reroute, it was just an idea of how it would function, one node is input, one node it output. It allows you to pipe output of one chain and have it output to like another group, several groups away, without wrapping lines all over crap. 

I'd assume it'd have a title, so you could customize it depending on your use.

**Comment by omar92** [comment]
i mean that you can route a start with and end normally , but to have option on settings to make gates routs invisible

**Comment by omar92** [comment]
there was update today that made the links have colors 

![image](https://user-images.githubusercontent.com/1458624/230745843-b6634f3d-655a-43e7-a621-f8f95ed8a2cd.png)

i think this can be used to do the gate thing by make gate have transparent output color , or same as bg

**Reply by ltdrdata** [reply]
We require a means of differentiating between whether it is transparent or unlinked.

Otherwise, gateways should always be created in pairs and destroyed in pairs.

And we still need a way to identify which gateway node is paired with which node.

**Reply by WASasquatch** [reply]
> there was update today that made the links have colors
> 
> ![image](https://user-images.githubusercontent.com/1458624/230745843-b6634f3d-655a-43e7-a621-f8f95ed8a2cd.png)
> 
> i think this can be used to do the gate thing by make gate have transparent output color , or same as bg

This definitely adds part of the functionality. Though makes me wonder if there could be a config on nodes themselves with features like `Hide Output Wire` so you could do this with regular reroute if you really wanted.

**Comment by lithoso** [comment]
Just use diffus3 set get nodes like everyone else.
https://github.com/diffus3/ComfyUI-extensions/tree/main/setget
![2023-05-22 00_36_49](https://github.com/comfyanonymous/ComfyUI/assets/124527631/b55e3c60-bff3-4f19-8975-85f6783b4130)

**Reply by WASasquatch** [reply]
> Just use diffus3 set get nodes like everyone else. https://github.com/diffus3/ComfyUI-extensions/tree/main/setget ![2023-05-22 00_36_49](https://user-images.githubusercontent.com/124527631/239837436-b55e3c60-bff3-4f19-8975-85f6783b4130.png)

Ironically, most the community probably doesn't know about this. Lol So everyone else, isn't, likely. This is the big problem with Comfy's community is even a plugin list is hidden away, and for said plugin list, I have to be aware of the existence of something to add it.
